core-overlay.html

```diff
@@ -111,666 +124,373 @@
+          this._prepareRenderOpenedAsync = null;
+        });
+        this.enableElementListener(this.opened, window, 'resize',
+            'resizeHandler');
+
+        if (this.opened) {
+          // force layout so SD Polyfill renders
+          this.offsetHeight;
+          this.discoverDimensions();
+          // if we are showing, then take care when positioning
+          this.preparePositioning();
+          this.positionTarget();
+          this.updateTargetDimensions();
+          this.finishPositioning();
         }
-      }
-      this.fire('core-overlay-' + (this.opened ? 'open' : 'close') + 
-          '-completed');
-      this.applyFocus();
-    },
-
-    prepareBackdrop: function() {
-      if (this.backdrop && this.opened) {
-        if (!this.scrim.parentNode) {
-          document.body.appendChild(this.scrim);
-          this.scrim.style.zIndex = currentOverlayZ() - 1;
+      },
+
+      // tasks which cause the overlay to actually open; typically play an
+      // animation
+      renderOpened: function() {
+        // this.notifyResize();
+        this.renderBackdropOpened();
+        // var transition = this.getTransition();
+        // if (transition) {
+        //   transition.go(this, {opened: this.opened});
+        // } else {
+          this.transitionend();
+        // }
+      },
+
+      // finishing tasks; typically called via a transition
+      transitionend: function(e) {
+        // make sure this is our transition event.
+        if (e && e.target !== this) {
+          return;
+        }
+        this.transitioning = false;
+        if (!this.opened) {
+          this.resetTargetDimensions();
+          this.style.display = 'none';
+          this.completeBackdrop();
+          manager.removeOverlay(this);
         }
-        trackBackdrop(this);
-      }
-    },
 
-    renderBackdropOpened: function() {
-      if (this.backdrop && getBackdrops().length < 2) {
-        this.scrim.classList.toggle('core-opened', this.opened);
-      }
-    },
+        this.applyFocus();
 
-    completeBackdrop: function() {
-      if (this.backdrop) {
-        trackBackdrop(this);
-        if (getBackdrops().length === 0) {
-          this.scrim.parentNode.removeChild(this.scrim);
-        }
-      }
-    },
+        // complete all tasks before firing -completed event
+        this.fire('core-overlay-' + (this.opened ? 'open' : 'close') +
+            '-completed');
+      },
 
-    preparePositioning: function() {
-      this.target.style.transition = this.target.style.webkitTransition = 'none';
-      this.target.style.transform = this.target.style.webkitTransform = 'none';
-      this.target.style.display = '';
-    },
+      prepareBackdrop: function() {
+        if (this.backdrop && this.opened) {
+          if (!this._backdrop.parentNode) {
+            Polymer.dom(document.body).appendChild(this._backdrop);
+            this._backdrop.style.zIndex = manager.currentOverlayZ() - 1;
```

Comment:
Maybe the `manager` should have a method that allows you to apply a z-index without knowing details about the internal book-keeping that is going on?